tests/foreman/installer/test_installer.py

```diff
@@ -177,20 +182,60 @@ def sat_non_default_install(module_sat_ready_rhels):
         'enable-foreman-plugin-discovery',
         'foreman-proxy-plugin-discovery-install-images true',
     ]
-    install_satellite(module_sat_ready_rhels[1], installer_args, enable_fapolicyd=True)
-    sat = module_sat_ready_rhels[1]
-    sat.enable_ipv6_http_proxy()
-    sat.execute('dnf -y --disableplugin=foreman-protector install foreman-discovery-image')
-    return sat
+
+    def deploy(request=None, delay=300):
```

Comment:
This is not a good idea to dissolve fixture `module_sat_ready_rhels` into nested `deploy` function.

Reply:
Purpose is clear, see how the non default install sat is being used for sanity and non sanity.


BTW I shall be rebasing it with radeks latest merge.

Reply:
The code is now changed for efficiency and reduce code complexity. Please take another look.